### ISSUE TITLE
Service always returns 200 with CompressResponseAttribute

### ISSUE THREAD

**Issue by svantreeck** [ServiceStack/Issues#533]
When I add a `[CompressResponse]` attribute to my service, when an error is thrown, I can see that it is thrown via logging; however, the client receives a response with a 200 status code (and a serialized version of the error in it's ResponseStatus property).  I am expecting to get a 404/403/etc in this situation.

```
[CompressResponse]
public class ThingService : Service 
{
        
        public object Get(GetThing request)
        {
            var thing = repo.Get(request.Id);

            if (thing == null)
            {
                throw HttpError.NotFound($"Thing {request.Id} not found");
            }
            ...
        }
}
```

**Comment by mythz** [comment]
ouch, should now be fixed with [this commit](https://github.com/ServiceStack/ServiceStack/commit/aca3edc71233ed0d35116bb2575be11f4120a3fe) which is available from v4.5.9 that's [now on MyGet](http://docs.servicestack.net/myget)

thx for reporting!

**Comment by svantreeck** [comment]
Thanks for the quick fix!
